src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -166,20 +166,40 @@ final case class PurposeApiServiceImpl(
     val operationLabel = s"Retrieving Purpose $id"
     logger.info(operationLabel)
 
+    def isDraft(purpose: PurposeManagementDependency.Purpose): Boolean = {
+      val states = purpose.versions.map(_.state)
+      states.nonEmpty &&
+      states == Seq(PurposeManagementDependency.PurposeVersionState.DRAFT)
+    }
+
+    def analyzeRiskAnalysisForm(
```

Comment:
I know it's not doing only that, but I don't have better ideas
```suggestion
    def authorizeRiskAnalysisForm(
```